doc/source/reading.rst

```diff
@@ -286,6 +286,81 @@ are automatically added. Other possible coordinates you may see:
 
 * ``acq_time``: Instrument data acquisition time per scan or row of data.
 
+Reading data before they're available
+=====================================
+
+Normally, data need to exist before they can be read.  This requirement
+impacts data processing timeliness.  For data arriving in segments,
+Satpy can process each segment immediately as it comes in.
+This feature currently only works with the :mod:`~satpy.readers.fci_l1c_nc` reader.
+This is experimental and likely to be instable and might change.
+
+Consider a near real time data reception situation where FCI segments are
+delivered one by one.  Classically, to produce full disc imagery, users
+would wait for all needed segments to arrive, before they start processing
+any data by passing all segments to the :class:`~satpy.scene.Scene`.
+For a more timely imagery production, users can create the Scene, load the
+data, resample, and even call :meth:`~satpy.scene.Scene.save_datasets`
+(as long as ``compute=False``) before the data are complete.
+When they then trigger the computation, much of the overhead in Satpy
+internals has already been completed, and Satpy will process each segment
+as it comes in.
+
+To do so, Satpy caches a selection of data and metadata between segments
+and between repeat cycles.  Caching between segments happens in-memory
+and needs no preparation from the user, but where data are cached
+between repeat cycles, the user needs to create this cache first from
+a repeat cycle that is available completely::
+
+  >>> from satpy.readers import create_preloadable_cache
+  >>> create_preloadable_cache("fci_l1c_nc", fci_files)
```

Comment:
Maybe, but it would add complexity to the code.  Information collection from cache happens during scene creation and loading, not computation, so we would need to add extra waiting functionality just for this scenario (we cannot wait for computing, because the cache contains information needed by the latest when `Scene.load` gets called and the `FileHandler` gets to work).  Considering preloading is an advanced and experimental topic and the code is complex enough as it is, I would prefer to keep the burden of creating it on the user to not make the code more complex yet.

But I will look at improved logging/error messages if cache is absent.